### PR TITLE
Zdoom survey 20210911

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Introduce `gzdoom`, a A feature centric port for all Doom engine games, based on ZDoom

Package(s) Affected
-------------------

- `gzdoom`
- `zmusic`
- `qzdl`

Security Update?
----------------

Build Order
-----------

`zmusic` -> `gzdoom`

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by WhiredPlanck** [comment]
## Known Problems
On ARM64:
```
[ERROR]: Found a loop in the dependency graph: vim, sane-backends, gnome-desktop, vala, colord-gtk, colord, gtk-3, qt-4, cups, ghostscript, graphviz, python-graphviz, automat, twisted, avahi
[ERROR]: Found a loop in the dependency graph: espeak, speech-dispatcher, orca, libcanberra, gdm, ffmpeg, pulseaudio
```

**UPDATED**: Fixed. Due to some new version dependencies doesn't exist in the AOSC repository at this moment.

**Comment by WhiredPlanck** [comment]
Look awesome (gzdoom, amd64)

image

File: extra-games/gzdoom/autobuild/beyond [hunk 1]
```diff
@@ -0,0 +1,19 @@
+abinfo 'Generating and installing .desktop file...'
+# adapted from AUR
+mkdir -pv "$PKGDIR"/usr/share/applications
+cat << EOF > "$PKGDIR"/usr/share/applications/gzdoom.desktop
+[Desktop Entry]
+Type=Application
+Name=GZDoom
+Comment=Advanced Doom source port with OpenGL support
```
Comment (MingcongBai) [inline extra-games/gzdoom/autobuild/beyond]: Drop "Advanced."

File: extra-games/gzdoom/autobuild/beyond [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Exec=gzdoom %F
+Terminal=false
+StartupWMClass=GZDoom
+MimeType=application/x-doom-wad;
+Categories=Game;ActionGame;
+EOF
+
+abinfo 'Installing icon...'
+install -Dvm644 "$SRCDIR"/src/posix/zdoom.xpm \
+                "$PKGDIR"/usr/share/icons/hicolor/scalable/apps/gzdoom.xpm
```
Comment (MingcongBai) [inline extra-games/gzdoom/autobuild/beyond]: This should go to `/usr/share/pixmaps/`.

File: extra-games/gzdoom/autobuild/beyond [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Type=Application
+Name=GZDoom
+Comment=Advanced Doom source port with OpenGL support
+Icon=gzdoom
+Exec=gzdoom %F
+Terminal=false
+StartupWMClass=GZDoom
+MimeType=application/x-doom-wad;
+Categories=Game;ActionGame;
+EOF
```
Comment (MingcongBai) [inline extra-games/gzdoom/autobuild/beyond:17]: You might want to consider translating this file.

File: extra-games/gzdoom/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,12 @@
+PKGNAME=gzdoom
+PKGSEC=games
+PKGDES="A feature centric port for all Doom engine games, based on ZDoom"
```
Comment (MingcongBai) [inline extra-games/gzdoom/autobuild/defines]: What does "feature centric" even mean?

File: extra-games/gzdoom/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+PKGDEP="sdl2 zlib bzip2 libjpeg-turbo fluidsynth game-music-emu openal-soft \
+        mpg123 libsndfile gtk-3 timidity++ nasm mesa glu tar sdl glew zmusic \
+        qzdl"
+BUILDDEP="git gcc"
+PKGPROV="zdoom"
+
+ABTYPE=cmake
+CMAKE_AFTER="-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+NOLTO=1
```
Comment (MingcongBai) [inline extra-games/gzdoom/autobuild/defines:12]: Why?

File: extra-games/gzdoom/autobuild/defines [hunk 6]
```diff
@@ -0,0 +1,12 @@
+PKGNAME=gzdoom
+PKGSEC=games
+PKGDES="A feature centric port for all Doom engine games, based on ZDoom"
+PKGDEP="sdl2 zlib bzip2 libjpeg-turbo fluidsynth game-music-emu openal-soft \
+        mpg123 libsndfile gtk-3 timidity++ nasm mesa glu tar sdl glew zmusic \
+        qzdl"
+BUILDDEP="git gcc"
+PKGPROV="zdoom"
+
+ABTYPE=cmake
```
Comment (MingcongBai) [inline extra-games/gzdoom/autobuild/defines]: Why not `cmakeninja`?

File: extra-games/qzdl/autobuild/defines [hunk 7]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=qzdl
+PKGSEC=games
+PKGDES="A popular launcher for GZDoom with user friendly interface"
```
Comment (MingcongBai) [inline extra-games/qzdl/autobuild/defines]: Note that your description is not exactly objective or descriptive (hint, "user friendly" is not exactly helpful).

File: extra-games/qzdl/autobuild/defines [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+PKGDEP="qt-5"
+BUILDDEP="git"
```
Comment (MingcongBai) [inline extra-games/qzdl/autobuild/defines]: This is not needed.

File: extra-games/qzdl/autobuild/defines [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+
+ABTYPE=cmake
```
Comment (MingcongBai) [inline extra-games/qzdl/autobuild/defines]: Why not `cmakeninja`...? Why did you have to specify this in the first place?

Comment (MingcongBai) [inline extra-games/qzdl/autobuild/defines]: Why not `cmakeninja`...? Why did you have to specify this in the first place?

File: extra-games/qzdl/autobuild/patch [hunk 10]
```diff
@@ -0,0 +1,3 @@
+abinfo 'Patching CMakeList.txt to install binaries automatically...'
+sed -e '53a install(TARGETS zdl DESTINATION ${CMAKE_INSTALL_BINDIR})' \
```
Comment (MingcongBai) [inline extra-games/qzdl/autobuild/patch]: Use a patch instead, sed is unreliable and can silently fail.

File: extra-games/zmusic/autobuild/beyond [hunk 11]
```diff
@@ -0,0 +1,3 @@
+abinfo 'Installing the licenses...'
+install -Dvm644 "$SRCDIR"/licenses/{bsd,dumb,legal,zmusic}.txt \
+        -t "$PKGDIR"/usr/share/doc/zmusic/licenses
```
Comment (MingcongBai) [inline extra-games/zmusic/autobuild/beyond]: Put it in `/usr/share/doc/zmusic/`, not in a subdirectory.

File: extra-games/zmusic/autobuild/defines [hunk 12]
```diff
@@ -0,0 +1,6 @@
+PKGNAME=zmusic
+PKGSEC=libs
+PKGDES="GZDoom's music system as a standalone library"
```
Comment (MingcongBai) [inline extra-games/zmusic/autobuild/defines]: What? This is a music system, for GZDoom?

File: extra-games/zmusic/autobuild/defines [hunk 13]
```diff
@@ -0,0 +1,6 @@
...
+PKGDEP="fluidsynth game-music-emu mpg123 libsndfile zlib alsa-lib"
+
+ABTYPE=cmake
```
Comment (MingcongBai) [inline extra-games/zmusic/autobuild/defines]: Again...

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-games/gzdoom/autobuild/defines (hunk 4 above)
Comment (WhiredPlanck) [reply extra-games/gzdoom/autobuild/defines]: This description was copied from upstream's, I think they want to say 'multi-features'......

File: extra-games/zmusic/autobuild/defines (hunk 12 above)
Comment (WhiredPlanck) [reply extra-games/zmusic/autobuild/defines]: Yes ... They make the ZMusic a standalone library. Although I found a binary of ZMusic in GZDoom source code package, I have no idea if I should use it.

File: extra-games/qzdl/autobuild/defines [hunk 14]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=qzdl
+PKGSEC=games
+PKGDES="A popular launcher for GZDoom using Qt"
```
Comment (MingcongBai) [inline extra-games/qzdl/autobuild/defines]: Just say "A launcher for GZDoom."

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-games/qzdl/autobuild/defines (hunk 14 above)
Comment (WhiredPlanck) [reply extra-games/qzdl/autobuild/defines]: OK, will fix soon.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**